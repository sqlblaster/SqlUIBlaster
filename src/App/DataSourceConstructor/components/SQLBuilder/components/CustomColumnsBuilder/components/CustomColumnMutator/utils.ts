import { ColumnBranch } from '../../../common/ColumnSelector/ColumnItem/SelectedColumn.models';
import { CustomColumn } from '../CustomColumnItem/model';

export const filterCustomColumn = (
  customColumnBranch: ColumnBranch,
  customColumns: CustomColumn[]
): boolean =>
  !customColumns.find(({ column }) => column.equals(customColumnBranch)) &&
  customColumnBranch !== customColumnBranch.lastColumn;
