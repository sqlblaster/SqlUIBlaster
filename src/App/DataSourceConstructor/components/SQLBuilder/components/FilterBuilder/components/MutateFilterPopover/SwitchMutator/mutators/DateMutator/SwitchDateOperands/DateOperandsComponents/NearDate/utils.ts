import { PluralDateComponent } from './plural-date-components';

export const getThisDateComponentNotation = (
  dateComponent: PluralDateComponent
) => {
  const singularForm = dateComponent.substr(0, dateComponent.length - 1);

  const thisDateNotation =
    singularForm === 'Day' ? 'Today' : `This ${singularForm}`;

  return thisDateNotation.toLowerCase();
};
