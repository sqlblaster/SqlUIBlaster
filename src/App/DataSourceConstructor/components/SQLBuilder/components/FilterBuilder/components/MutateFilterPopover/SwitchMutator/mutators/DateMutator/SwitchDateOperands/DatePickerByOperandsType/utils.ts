import { SingleDateOperands } from '../DateOperandsComponents/SingleDate/model';
import { FixedDateOperands } from './components/FixedDatePicker/model';
import { RelativeDateOperands } from './components/RelativeDatePicker/model';
import { VariableDateOperands } from './components/VariableDatePicker/model';
import { DateType } from './date-operand-types';

export const getNewOperands = (dateType: DateType): SingleDateOperands => {
  switch (dateType) {
    case 'Fixed date':
      return new FixedDateOperands();
    case 'Relative date':
      return new RelativeDateOperands();
    case 'Variable date':
      return new VariableDateOperands();
    default:
      throw new Error('unknown date type is provided');
  }
};
