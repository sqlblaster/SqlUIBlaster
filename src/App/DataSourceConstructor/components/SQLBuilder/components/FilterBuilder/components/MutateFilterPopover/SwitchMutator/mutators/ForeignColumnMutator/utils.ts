import { ForeignColumnMutatorProps } from './props';

export const getModels = (
  props: Pick<ForeignColumnMutatorProps, 'enums' | 'column'>
): string[] | undefined => {
  const {
    enums,
    column: {
      lastColumn: { foreignModelName, modelName, type }
    }
  } = props;

  return type === 'uuid' && !enums
    ? undefined
    : type === 'uuid' && enums
    ? enums
    : type === 'primary'
    ? [modelName]
    : foreignModelName
    ? [foreignModelName]
    : undefined;
};
