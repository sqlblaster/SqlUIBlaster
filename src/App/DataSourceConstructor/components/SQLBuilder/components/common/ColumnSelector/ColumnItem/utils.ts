import { clone } from 'ramda';
import { ColumnBranch } from './SelectedColumn.models';

export const getNewTableBranch = (
  parentColumnBranch: ColumnBranch,
  foreignTableName: ColumnBranch['foreignTableName'],
  foreignModelName: ColumnBranch['foreignModelName']
) => {
  const newTableBranch = clone(parentColumnBranch);

  newTableBranch.lastColumn.foreignColumn = new ColumnBranch({
    tableName: foreignTableName,
    modelName: foreignModelName
  });

  newTableBranch.lastColumn = newTableBranch.lastColumn.foreignColumn;

  return newTableBranch;
};

export const listItemOwnPaddingLeft: number = 16;
export const paddingShiftAmountByLevel: number = 15;
export const calcPaddingAccordingTo = (level: number) => {
  return listItemOwnPaddingLeft + level * paddingShiftAmountByLevel;
};
