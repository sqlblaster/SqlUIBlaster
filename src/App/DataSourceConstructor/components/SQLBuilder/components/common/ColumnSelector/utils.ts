import { clone } from 'ramda';
import { Column } from 'src/App/DataSourceConstructor/schemas';
import { ColumnBranch } from './ColumnItem/SelectedColumn.models';
import { ColumnSelectorOwnProps } from './props';

export const getRootTableBranch = (tableName: string, modelName: string) =>
  new ColumnBranch({ tableName, modelName });

export const getNewColumnBranch = (
  tableBranch: ColumnBranch,
  { name, type, foreignTableName, foreignModelName }: Column
) => {
  const newColumnBranch = new ColumnBranch(clone(tableBranch));

  const lastColumn = newColumnBranch.lastColumn;
  lastColumn.columnName = name;
  lastColumn.type = type;
  lastColumn.foreignTableName = foreignTableName;
  lastColumn.foreignModelName = foreignModelName;

  return newColumnBranch;
};

export const sortInOrderForeignColumnsBelow = (columns: Column[]) =>
  columns.sort((c1, c2) => +!!c1.foreignTableName - +!!c2.foreignTableName);

export const filterColumns = (
  columns: Column[],
  canShowColumnBranch: Required<ColumnSelectorOwnProps>['canShowColumnBranch'],
  tableBranch: ColumnBranch
): Column[] => {
  return columns.filter(
    column =>
      canShowColumnBranch(getNewColumnBranch(tableBranch, column)) ||
      column.foreignTableName
  );
};
