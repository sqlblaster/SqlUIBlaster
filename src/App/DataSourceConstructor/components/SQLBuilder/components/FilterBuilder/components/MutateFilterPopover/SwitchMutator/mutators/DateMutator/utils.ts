import { DateOperation } from './DateOperation';
import { CoupleDateOperation } from './SwitchDateOperands/DateOperandsComponents/CoupleDates/model';
import { EmptyDateOperation } from './SwitchDateOperands/DateOperandsComponents/EmptyDates/model';
import { NearDateOperation } from './SwitchDateOperands/DateOperandsComponents/NearDate/model';
import { SingleDateOperation } from './SwitchDateOperands/DateOperandsComponents/SingleDate/model';

export const getDateOperationVia = (
  operator: DateOperation['operator']
): DateOperation => {
  switch (operator) {
    case 'Previous':
    case 'Next':
      return new NearDateOperation({ operator });
    case 'Before':
    case 'After':
    case 'On':
      return new SingleDateOperation({ operator });
    case 'Between':
      return new CoupleDateOperation({ operator });
    case 'Is empty':
    case 'Is not empty':
    default:
      return new EmptyDateOperation({ operator });
  }
};
