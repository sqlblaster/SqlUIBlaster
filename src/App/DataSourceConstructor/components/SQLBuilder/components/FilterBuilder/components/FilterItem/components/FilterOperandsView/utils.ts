import { Operation } from '../../../MutateFilterPopover/Operation';
import { DateOperation } from '../../../MutateFilterPopover/SwitchMutator/mutators/DateMutator/DateOperation';
import { CoupleOperandsViewProps } from './components/CoupleOperandsView';
import { SingleOperandViewProps } from './components/SingleOperandView';
import {
  formatCoupleDateOperands,
  formatNearDateOperands,
  formatSingleDateOperands
} from './date-formatter';

export const hasOperands = (operator: Operation['operator']): boolean => {
  switch (operator) {
    case 'Is empty':
    case 'Is not empty':
    case 'Is true':
    case 'Is false':
      return false;
    default:
      return true;
  }
};

export const getFormattedDateOperand = (operation: DateOperation): string => {
  switch (operation.operator) {
    case 'Previous':
    case 'Next':
      return formatNearDateOperands(operation.operands, operation.operator);
    case 'Before':
    case 'After':
    case 'On':
      return formatSingleDateOperands(operation.operands);
    default:
      return '';
  }
};

export const getFormattedCoupleOperands = (
  operation: Operation
): CoupleOperandsViewProps['operands'] => {
  if (operation.type === 'Date' && operation.operands instanceof Array) {
    return formatCoupleDateOperands(operation.operands);
  } else if (
    operation.type === 'Number' &&
    operation.operands instanceof Array
  ) {
    return operation.operands;
  } else {
    throw new Error('expected between operands');
  }
};

export const getFormattedSingleOperand = (
  operation: Operation
): SingleOperandViewProps['operand'] => {
  if (
    hasOperands(operation.operator) &&
    !(operation.operands instanceof Array) &&
    hasOperands(operation.operator)
  ) {
    switch (operation.type) {
      case 'Foreign': {
        const record = operation.operands;

        return record && record.title ? record.title : '';
      }
      case 'Date':
        return getFormattedDateOperand(operation);
      case 'Number':
      case 'String':
      default:
        return operation.operands as number | string;
    }
  } else {
    throw new Error('expected singular operand');
  }
};
