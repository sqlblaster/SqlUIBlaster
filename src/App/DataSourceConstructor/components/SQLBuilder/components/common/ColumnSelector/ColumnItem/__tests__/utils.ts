import { ColumnBranch } from '../SelectedColumn.models';
import { calcPaddingAccordingTo, getNewTableBranch } from '../utils';

describe('getNewTableBranch', () => {
  it('should create new table branch on the basis of a column branch', () => {
    const columnName = 'ProductID';
    const columnBranch = new ColumnBranch({
      columnName
    });
    const foreignTableName = 'Products';

    const tableBranch = getNewTableBranch(
      columnBranch,
      foreignTableName,
      foreignTableName
    );

    expect(tableBranch).toMatchSnapshot();
  });
});

describe('calcPaddingAccordingTo', () => {
  it('should work properly', () => {
    expect(calcPaddingAccordingTo(1)).toBe(31);
    expect(calcPaddingAccordingTo(3)).toBe(61);
  });
});
