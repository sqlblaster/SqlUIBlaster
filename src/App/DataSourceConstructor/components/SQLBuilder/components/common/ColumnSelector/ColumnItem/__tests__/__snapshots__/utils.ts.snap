// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNewTableBranch should create new table branch on the basis of a column branch 1`] = `
Object {
  "_foreignTableName": undefined,
  "_lastColumn": undefined,
  "columnName": "ProductID",
  "foreignColumn": Object {
    "_foreignTableName": undefined,
    "_lastColumn": undefined,
    "columnName": "",
    "foreignColumn": undefined,
    "foreignModelName": undefined,
    "foreignTableName": undefined,
    "tableName": "Products",
    "type": undefined,
  },
  "foreignModelName": undefined,
  "foreignTableName": undefined,
  "tableName": "",
  "type": undefined,
}
`;
